pkg/limayaml/defaults.go

```diff
@@ -284,7 +284,7 @@ func FillDefault(y, d, o *LimaYAML, filePath string, warn bool) {
 	}
 	y.Arch = ptr.Of(ResolveArch(y.Arch))
 
-	y.Images = append(append(o.Images, y.Images...), d.Images...)
+	y.Images = slices.Concat(o.Images, y.Images, d.Images)
```

Comment:
Thanks! It is not only "clearer and more concise" but also more correct:

The original code would modify `o.Images` but assign the result to `y.Images`. Which doesn't matter if `o.Images` is not going to be re-used. But it is an undesired side-effect, which is not present in `slices.Concat`.

Reply:
> I wonder if there is some linter option that could enforce that the first argument to append must match the variable the result is being assigned to. Well, and that handles nested append calls too. So to enforce that a == x in an expression like:

[gocritic.appendAssign](https://go-critic.com/overview.html#appendassign) might detect that

Reply:
It seems like it doesn't since we already have it enabled, and it never flagged the previous code.

Maybe not that important since we don't have any nested `append` calls anymore at all.

Reply:
Created issue https://github.com/go-critic/go-critic/issues/1463 for that.